core/src/main/java/org/mapfish/print/http/ConfigFileResolvingHttpRequestFactory.java

```diff
@@ -157,6 +150,89 @@ protected synchronized ClientHttpResponse executeInternal(final HttpHeaders head
             }
         }
 
+        private ClientHttpResponse doDataUriRequest() throws IOException {
+            final String urlStr = this.uri.toString();
+            final URL url = new URL("data", null, 0,
+                urlStr.substring("data:".length()), new Handler());
+            final DataUrlConnection duc = new DataUrlConnection(url);
+            final InputStream is = duc.getInputStream();
+            final String contentType = duc.getContentType();
+            final HttpHeaders responseHeaders = new HttpHeaders();
+            responseHeaders.set("Content-Type", contentType);
+            final ConfigFileResolverHttpResponse response =
+            new ConfigFileResolverHttpResponse(is, responseHeaders);
+            LOGGER.debug("Resolved request using DataUrlConnection: {}", contentType);
+            return response;
+        }
+
+        private ClientHttpResponse doFileRequest() throws IOException {
+            final String uriString = this.uri.toString();
+            final Configuration configuration = ConfigFileResolvingHttpRequestFactory.this.config;
+            final byte[] bytes = configuration.loadFile(uriString);
+            final InputStream is = new ByteArrayInputStream(bytes);
+            final HttpHeaders responseHeaders = new HttpHeaders();
+            final Optional<File> file = configuration.getFile(uriString);
+            if (file.isPresent()) {
+                responseHeaders.set(
+                    "Content-Length", String.valueOf(Files.probeContentType(file.get().toPath()))
+                );
+            }
+            final ConfigFileResolverHttpResponse response =
+                    new ConfigFileResolverHttpResponse(is, responseHeaders);
+            LOGGER.debug("Resolved request: {} using MapFish print config file loaders.",
+                            uriString);
+            return response;
+        }
+
+        private ClientHttpResponse doHttpRequestWithRetry(final HttpHeaders headers) throws IOException {
+            AtomicInteger counter = new AtomicInteger();
```

Comment:
The refactoring and solution but I could not work out if any of the tests validated that the retry worked.

Reply:
I don't know how to test that in Java, for that, we should be able to mock the result of the HTTP request...